### ISSUE TITLE
How would you apply a patch to all imports of a given file

### ISSUE THREAD

**Issue by brendon** [Envisage/patch-loader#1]
I'm attempting to patch the main file for a large library. It has a large number of plugins that need to be imported after importing the main constructor. These plugin files require the main file also. I suspect that because these plugin files (not written by me) don't import the patched version, I'm getting two copies of the constructor. I notice this because some customisation I make to the constructor before re-exporting it doesn't stick.

I guess the question is, how would one force this patch to be applied any time the source file is imported?

**Comment by jiggak** [comment]
That sounds like a pickle. Are you able to share which library you are importing, and what the import looks like?

**Comment by brendon** [comment]
I can do one better, as I figured it out. :D I'm using Froala Editor. This is `webpacker` configuration but you'll get the gist:

```
const { environment } = require('@rails/webpacker')

environment.config.merge({
  resolve: {
    alias: {
      'froala-editor': 'froala-editor/js/froala_editor'
    }
  }
})

environment.loaders.append('patch', {
  test: require.resolve('froala-editor/js/froala_editor'),
  loader: 'patch-loader',
  options: {
    patch: '../../../patches/froala-editor/froala_editor.js.patch'
  }
})

```

Firstly I get access to the froala source files through my subscription, but their package.json still points at the minified ones (thus the alias). Then I did as you suggested would probably not be needed in your `README` and used the object style configuration instead which affects all cases when `froala-editor/js/froala_editor` is required (even if it's via the `froala-editor` alias via it's own plugin files).

Works a treat! :D Thanks for getting back to me though. Proud to be your first issue! Thanks for making the package :)
